src/nnet3/nnet-training.cc

```diff
@@ -78,6 +78,7 @@ void NnetTrainer::ProcessOutputs(const NnetExample &eg,
                                  NnetComputer *computer) {
   std::vector<NnetIo>::const_iterator iter = eg.io.begin(),
       end = eg.io.end();
+  num_minibatches_processed_++;
```

Comment:
this statement should be after the loop, not before (otherwise it might change the function's behavior in the normal case).. ++ is post-increment, not pre-increment.